src/core/libmaven/Compound.cpp

```diff
@@ -72,17 +79,35 @@ bool Compound::operator == (const Compound& rhs) const
             && type() == rhs.type());
 }
 
-float Compound::adjustedMass(int charge) { 
-     /**   
-    *@return    -    total mass by formula minus loss of electrons' mass 
+
+void Compound::operator = (const Compound& rhs)
+{
+    _id = rhs._id;
+    _name = rhs._name;
+    _formula = rhs._formula;
+    _srmId = rhs._srmId;
+    _expectedRt = rhs._expectedRt;
+    _charge = rhs._charge;
+    _mass = rhs._mass;
+    _precursorMz = rhs._precursorMz;
+    _productMz = rhs._productMz;
+    _collisionEnergy = rhs._collisionEnergy;
+    category = rhs.category;
+}
```

Comment:
These are not the only attributes that should be copied to give you an identical `Compound` object. Take a look at all the properties being compared in the `==` operator above. All of those should be here, at least.

Reply:
This one still needs attention.

Reply:
I made the change, assigned all attributes. But I will still have a look